command/agent/config_parse_test.go

```diff
@@ -904,3 +946,50 @@ func permutations(arr []string) [][]string {
 	helper(arr, len(arr))
 	return res
 }
+
+func TestConfig_MultipleVault(t *testing.T) {
```

Comment:
Maybe I missed, but would it be worth to have tests where the default `vault` block doesn't have a name and that we test not only that `Vault` is equal to `Vaults["default"]` but that they actually point to the same object?

Reply:
Good call, we could test that by removing the `name` field from the default Vault in `command/agent/testdata/extra-vault.hcl`. And we can check that the pointers are the same.

Reply:
Fixed in 3dcd0a7913d341bb6bbde1018202bf274c9f3173